src/FormElement/ScheduleElement.php

```diff
@@ -0,0 +1,314 @@
+    {
+        $repeat = $this->getFrequency();
+        $interval = $this->getValue('interval', 1);
+        switch ($repeat) {
+            case RRule::DAILY:
+                if ($interval === '*') {
+                    $interval = 1;
+                }
+
+                return new RRule("FREQ=DAILY;INTERVAL=$interval");
+            case RRule::WEEKLY:
+                $byDay = implode(',', $this->weeklyField->getSelectedWeekDays());
+
+                return new RRule("FREQ=WEEKLY;INTERVAL=$interval;BYDAY=$byDay");
+            /** @noinspection PhpMissingBreakStatementInspection */
+            case RRule::MONTHLY:
+                $runsOn = $this->monthlyFields->getValue('runsOn', MonthlyFields::RUNS_EACH);
+                if ($runsOn === MonthlyFields::RUNS_EACH) {
+                    $byMonth = implode(',', $this->monthlyFields->getSelectedDays());
+
+                    return new RRule("FREQ=MONTHLY;INTERVAL=$interval;BYMONTHDAY=$byMonth");
+                }
+            // Fall-through to the next switch case
+            case RRule::YEARLY:
+                $rule = "FREQ=MONTHLY;INTERVAL=$interval;";
+                if ($repeat === RRule::YEARLY) {
+                    $runsOn = $this->annuallyFields->getValue('runsOnThe', 'n');
+                    $month = $this->annuallyFields->getValue('month', (int) $this->start->format('m'));
+                    if (is_string($month)) {
+                        $datetime = DateTime::createFromFormat('!M', $month);
+                        if (! $datetime) {
+                            throw new InvalidArgumentException(sprintf('Invalid month provided: %s', $month));
+                        }
+
+                        $month = (int) $datetime->format('m');
+                    }
+
+                    $rule = "FREQ=YEARLY;INTERVAL=1;BYMONTH=$month;";
+                    if ($runsOn === 'n') {
+                        return new RRule($rule);
+                    }
+                }
+
+                $element = $this->monthlyFields;
+                if ($repeat === RRule::YEARLY) {
+                    $element = $this->annuallyFields;
+                }
+
+                $runDay = $element->getValue('day', $element::$everyDay);
+                $ordinal = $element->getValue('ordinal', $element::$first);
+                $position = $element->getOrdinalAsInteger($ordinal);
+
+                if ($runDay === $element::$everyDay) {
+                    $rule .= "BYDAY=MO,TU,WE,TH,FR,SA,SU;BYSETPOS=$position";
+                } elseif ($runDay === $element::$everyWeekday) {
+                    $rule .= "BYDAY=MO,TU,WE,TH,FR;BYSETPOS=$position";
+                } elseif ($runDay === $element::$everyWeekend) {
+                    $rule .= "BYDAY=SA,SU;BYSETPOS=$position";
+                } else {
+                    $rule .= sprintf('BYDAY=%d%s', $position, $runDay);
+                }
+
+                return new RRule($rule);
+        }
+        // Oops!!
```

Comment:
What's this? Is this a reason for an error? Then throw an exception.

Reply:
I don't think that will ever happen! But since I don't have a default switch case, PhpStorm renders an error hint.

Reply:
Ah, well, this hint. I've disabled it for this reason :sweat_smile: 

But even worse is that if it occurs, PHP complains with a cryptic (for the user) error message as the return type must not be null.

Just add a `default` case and throw a `LogicException` there. Then PHPStorm and PHP are happy.